helm/templates/storage/pvc.yaml

```diff
@@ -1,11 +1,28 @@
-apiVersion: v1
+{{- if and .Values.persistence.enabled .Values.persistence.storageData.enabled (not .Values.persistence.storageData.existingClaim) }}
+---
 kind: PersistentVolumeClaim
+apiVersion: v1
 metadata:
-  name: {{.Values.storage.pvc.name}}
+  name: {{ template "sbombastic.fullname" . }}-storage-data
+  labels:
+  {{- with .Values.persistence.storageData.labels }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  annotations:
+  {{- with .Values.persistence.storageData.annotations }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
-  storageClassName: {{ .Values.storage.storageClass.name }}
   accessModes:
-    {{ .Values.storage.pvc.accessModes | toYaml | nindent 2 }}
+    - {{ .Values.persistence.storageData.accessMode | quote }}
   resources:
     requests:
-      storage: {{ .Values.storage.pvc.requestStorage }}
\ No newline at end of file
+      storage: {{ .Values.persistence.storageData.size | quote }}
+  {{- with .Values.persistence.storageData.storageClass }}
+  {{- if (eq "-" .) }}
+  storageClassName: standard
+  {{- else }}
+  storageClassName: "{{ . }}"
+  {{- end }}
```

Comment:
the `standard` storage class is not granted. Many k8s distro have it, but it's not guaranteed to be present.
Using an empty name is safer, since it will fall back to use whatever storage class has been marked as the default one by the admin.

Finally, I'm not super convinced about this handling of the `-` symbol. I would just propagate whatever the user put into the `values`. However, if you want to keep the `-` thing around, at least document that inside of the values.yaml